dashboard/app/jobs/application_job.rb

```diff
@@ -1,7 +1,71 @@
+require 'cdo/aws/metrics'
+
 class ApplicationJob < ActiveJob::Base
   # Automatically retry jobs that encountered a deadlock
   # retry_on ActiveRecord::Deadlocked
 
   # Most jobs are safe to ignore if the underlying records are no longer available
   # discard_on ActiveJob::DeserializationError
+
+  METRICS_NAMESPACE = 'code-dot-org/ActiveJob'
+
+  def self.common_dimensions
+    [
+      {
+        name: 'Environment',
+        value: CDO.rack_env
+      },
+      {
+        name: 'JobName',
+        value: name
+      }
+    ]
+  end
```

Comment:
This and the code that calls it, `job.class.common_dimensions` is some Ruby inheritance logic that I don't fully understand. If there's a better way to deduplicate this code while still referencing the name of the subclass, I'm all for it.

Reply:
not sure I understand exactly what you don't like about your current implementation, but if you'd prefer you could convert this from a class method to an instance method.

Reply:
I was looking for the best way to deduplicate this array out of each of the metrics arrays below. If this is a suitable way, that's fine.

Reply:
I merged in some changes to the cron job and added a note that they publish to the same metric.